spec/unit/iiif/v3/presentation/service_spec.rb

```diff
@@ -1,28 +1,150 @@
 describe IIIF::V3::Presentation::Service do
 
-  describe 'self#get_descendant_class_by_jld_type' do
-    before do
-      class DummyClass < IIIF::V3::Presentation::Service
-        TYPE = "Collection"
-        def self.singleton_class?
-          true
-        end
+  describe '#required_keys' do
+    it '"type" is not required' do
+      expect(subject.required_keys).not_to include('type')
+    end
+  end
+
+  describe '#prohibited_keys' do
+    keys = IIIF::V3::Presentation::Service::CONTENT_RESOURCE_PROPERTIES +
+      IIIF::V3::Presentation::Service::PAGING_PROPERTIES +
+      %w{
+        nav_date
+        viewing_direction
+        start_canvas
+        content_annotation
+      }
+    keys.each do |k|
```

Comment:
I like your suggestion;  showing each individual key makes sense when it's a small list with required keys;  less so in this context.  I'm not a big splat user so it didn't occur to me.

I also found rspec has a "contains_exactly" matcher which would be perfect if we didn't want to leave the possibility open for the base class to have prohibited keys of its own.